src/trio/_path.py

```diff
@@ -32,219 +22,110 @@
-        if isinstance(other, cls):
-            other = cast(Path, other)._wrapped
-        value = attr(self._wrapped, other)
-        return rewrap_path(value)
+def _wrap_method(
+    fn: Callable[Concatenate[pathlib.Path, P], T],
+) -> Callable[Concatenate[Path, P], Awaitable[T]]:
+    @_wraps_async(fn)
+    def wrapper(self: Path, /, *args: P.args, **kwargs: P.kwargs) -> T:
+        return fn(self.wrapped_cls(self), *args, **kwargs)
 
-    assert isinstance(wrapper, types.FunctionType)
-    wrapper.__signature__ = inspect.signature(attr)
     return wrapper
 
 
-def iter_wrapper_factory(
-    cls: AsyncAutoWrapperType, meth_name: str
-) -> Callable[Concatenate[Path, P], Awaitable[Iterable[Path]]]:
-    @async_wraps(cls, cls._wraps, meth_name)
-    async def wrapper(self: Path, *args: P.args, **kwargs: P.kwargs) -> Iterable[Path]:
-        meth = getattr(self._wrapped, meth_name)
-        func = partial(meth, *args, **kwargs)
-        # Make sure that the full iteration is performed in the thread
-        # by converting the generator produced by pathlib into a list
-        items = await trio.to_thread.run_sync(lambda: list(func()))
-        return (rewrap_path(item) for item in items)
+def _wrap_method_path(
+    fn: Callable[Concatenate[pathlib.Path, P], pathlib.Path],
+) -> Callable[Concatenate[PathT, P], Awaitable[PathT]]:
+    @_wraps_async(fn)
+    def wrapper(self: PathT, /, *args: P.args, **kwargs: P.kwargs) -> PathT:
+        return self.__class__(fn(self.wrapped_cls(self), *args, **kwargs))
 
     return wrapper
 
 
-def thread_wrapper_factory(
-    cls: AsyncAutoWrapperType, meth_name: str
-) -> Callable[Concatenate[Path, P], Awaitable[Path]]:
-    @async_wraps(cls, cls._wraps, meth_name)
-    async def wrapper(self: Path, *args: P.args, **kwargs: P.kwargs) -> Path:
-        meth = getattr(self._wrapped, meth_name)
-        func = partial(meth, *args, **kwargs)
-        value = await trio.to_thread.run_sync(func)
-        return rewrap_path(value)
-
+def _wrap_method_path_iterable(
+    fn: Callable[Concatenate[pathlib.Path, P], Iterable[pathlib.Path]],
+) -> Callable[Concatenate[PathT, P], Awaitable[Iterable[PathT]]]:
+    @_wraps_async(fn)
+    def wrapper(self: PathT, /, *args: P.args, **kwargs: P.kwargs) -> Iterable[PathT]:
+        return map(self.__class__, [*fn(self.wrapped_cls(self), *args, **kwargs)])
+
+    assert wrapper.__doc__ is not None
+    wrapper.__doc__ += (
+        f"\n"
+        f"This is an async method that returns a synchronous iterator, so you\n"
+        f"use it like:\n"
+        f"\n"
+        f".. code:: python\n"
+        f"\n"
+        f"    for subpath in await mypath.{fn.__name__}():\n"
+        f"        ...\n"
+        f"\n"
+        f".. note::\n"
+        f"\n"
+        f"    The iterator is loaded into memory immediately during the initial\n"
+        f"    call, (see `issue #501\n"
+        f"    <https://github.com/python-trio/trio/issues/501>`__ for discussion).\n"
```

Comment:
```suggestion
        f"    The iterator is loaded into memory immediately during the initial\n"
        f"    call. (see `issue #501\n"
        f"    <https://github.com/python-trio/trio/issues/501>`__ for discussion)\n"
```

I think the comma is weird grammatically. Nitpick though

Reply:
Oh yea, I was thinking about this when I reviewed the other day, why are all of the strings in this block f-strings if only a few actually need it?

Reply:
the consistent `f` prefixes help text alignment. `textwrap.dedent` would probably be nicer but it's unnecessary and can have its quirks (w/r/t how it decides how much to dedent)

Reply:
> I think the comma is weird grammatically. Nitpick though

I think that creates a different grammatical problem (capitalization of `(see`.). So I checked with the _authority_ that is grammarly and apparently just removing the comma is correct.

https://www.grammarly.com/blog/comma-before-parenthesis-or-after/

So I did that.